infra/aws/terraform/prow-build-cluster/eks.tf

```diff
@@ -133,3 +163,29 @@ module "eks" {
     }
   }
 }
+
+# We create IAM role to delegate access to the cluster.
+resource "aws_iam_role" "iam_cluster_admin" {
+  name        = "Prow-Cluster-Admin"
+  description = "IAM role used to delegate access to prow-build-cluster"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "AWS" : "468814281478"
+        },
+        "Action" : "sts:AssumeRole",
+        "Condition" : {}
+      }
+    ]
+  })
+}
+
+# Give administrator access to the admin IAM role so it can be used with Terraform.
+resource "aws_iam_role_policy_attachment" "iam_policy_cluster_admin" {
+  role       = aws_iam_role.iam_cluster_admin.name
+  policy_arn = "arn:aws:iam::aws:policy/AdministratorAccess"
```

Comment:
This is more access than is needed. Consider a permissions boundary or a more specific policy.

For example, this allows `organizations:LeaveOrganization`. A deployer should not need to be allowed to do so.

Reply:
The problem here is how to determine what are needed permissions. This account should be able to deploy those Terraform configs, and it needs quite a lot of access, so going Administrator level made sense. If anyone has any idea how to properly collect needed permissions, it would be appreciated.

Reply:
Try https://skildops.com/blog/generate-restricted-aws-iam-policy-via-cloudtrail as a starting point.